service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseNoteServiceTest.java

```diff
@@ -0,0 +1,98 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.reform.fpl.model.CaseNote;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
+import uk.gov.hmcts.reform.idam.client.models.UserDetails;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.unwrapElements;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.wrapElements;
+
+@ExtendWith(MockitoExtension.class)
+class CaseNoteServiceTest {
+
+    @Mock
+    private IdamClient idamClient;
+
+    @InjectMocks
+    private CaseNoteService service;
+
+    private static final String userAuthToken = "Bearer";
+    private static final UserDetails userDetails = UserDetails.builder()
+        .forename("John")
+        .surname("Smith")
+        .build();
+
+    @Nested
+    class BuildCaseNote {
+
+        @BeforeEach
+        void setup() {
+            given(idamClient.getUserDetails(userAuthToken)).willReturn(userDetails);
+        }
```

Comment:
is this nested class + BeforeEach necessary? if there is only one test, can we put the given() at the start of that test and remove the class

Reply:
mocks are not relevant to the actual test so I don't want them clogging it up, before seems a better place

Reply:
nested is required as mock is not relevant to all test methods in class

Reply:
makes sense 👍